torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1009,7 +979,7 @@ std::shared_ptr<JitFuture> TensorPipeAgent::send(
                 // A read error will lead all following callbacks to be
                 // invoked with error, and shouldn't reach here.
                 TORCH_INTERNAL_ASSERT(
-                    !clientPipe.readError_, "Shouldn't in error state");
+                    !clientPipe.inError_, "Shouldn't be in error state");
                 auto it = clientPipe.pendingResponseMessage_.find(messageId);
```

Comment:
Would I be correct if I assume that, for the flushed Futures (which are already marked with error), we will never reach here again? 

Is it possible that although the `pipeWrite` failed, but the message was actually sent, and later a response might come back from the callee?

Reply:
For a given pipe, once an error occurs (in either read or write), everything will result in an error. Hence this code here will never be reached after we've hit an error (and thus flushed all futures) and hence this check won't fail.

Or, well, that's the intended behavior. If this check ever fails it will be due to a bug in TensorPipe.